packages/svelte/src/main/main-client.js

```diff
@@ -210,12 +188,15 @@ function init_update_callbacks() {
  * @returns {void}
  */
 export function beforeUpdate(fn) {
-	const component_context = current_component_context;
-	if (component_context === null) {
-		throw new Error('beforeUpdate can only be used during component initialisation.');
+	if (current_component_context === null) {
+		throw new Error('beforeUpdate can only be used during component initialisation');
+	}
+
+	if (current_component_context.r) {
+		throw new Error('beforeUpdate cannot be used in runes mode');
```

Comment:
Does it make sense to also throw a validation error in the compiler if it detects a `before/afterUpdate` import from `svelte`? That would probably catch most of these cases at compile time, so there's less surprises at runtime.

Reply:
could do, yeah — maybe as a follow-up PR?

Reply:
nvm, added it here, was an easy add